src/components/notifications/WebhookConfigModal.tsx

```diff
@@ -0,0 +1,372 @@
+        this.setState({ form });
+    }
+
+    isValid(event, key: 'configName' | 'webhookUrl' | 'payload'): void {
+        let { form, isValid } = { ...this.state };
+        if (key != 'payload') {
+            isValid[key] = event.target.value.length !== 0;
+        } else if (this.state.form.payload != '') {
+            try {
+                JSON.parse(this.state.form.payload)
+                isValid[key] = true
+            } catch (err) {
+                isValid[key] = false
+            }
+        }
+        this.setState({ form, isValid });
+    }
+
+    handleWebhookUrlChange(event: React.ChangeEvent<HTMLInputElement>): void {
+        let { form } = { ...this.state };
+        form.webhookUrl = event.target.value;
+        this.setState({ form });
+    }
+
+    handleWebhookPaylodChange(value): void {
+        let { form } = { ...this.state };
+        form.payload = value;
+        this.setState({ form });
+    }
+
+    saveWebhookConfig(): void {
+        let state = { ...this.state };
+        state.form.isLoading = true;
+        this.setState(state);
+        let keys = Object.keys(this.state.isValid);
+        let isFormValid = keys.reduce((isFormValid, key) => {
+            isFormValid = isFormValid && this.state.isValid[key];
+            return isFormValid;
+        }, true);
+        if (!isFormValid) {
+            state.form.isLoading = false;
+            state.form.isError = true;
+            this.setState(state);
+            return;
+        }
+        let requestBody = this.state.form;
+        if (this.props.webhookConfigId) requestBody['id'] = this.props.webhookConfigId;
+        let promise = this.props.webhookConfigId ? updateWebhookConfiguration(requestBody) : saveWebhookConfiguration(requestBody);
+        promise.then((response) => {
+            let state = { ...this.state };
+            state.form.isLoading = false;
+            state.form.isError = false;
+            this.setState(state);
+            toast.success("Saved Successfully");
+            this.props.onSaveSuccess();
+        }).catch((error) => {
+            showError(error);
+        })
+    }
+
+    setHeaderData(index, _headerData) {
+        const _headers = [...this.state.form.header]
+        _headers[index] = _headerData
+        let { form } = { ...this.state };
+        form.header = _headers;
+        this.setState({ form });
+    }
+
+    addNewHeader() {
+        const _headers = [...this.state.form.header]
+        _headers.splice(0, 0, {
+            key: '',
+            value: '',
+        })
+        let { form } = { ...this.state };
+        form.header = _headers;
+        this.setState({ form });
+    }
+
+    removeHeader(index) {
+        const _headers = [...this.state.form.header]
+        _headers.splice(index, 1)
+        let { form } = { ...this.state };
+        form.header = _headers;
+        this.setState({ form });
+    }
+
+    setCopied(value: boolean) {
+        let { copyAttribute } = { ...this.state }
```

Comment:
we don't need to fetch this from set. We can directly set the new value in setState

Reply:
done